vyos/vyos-1x
# T4767: refactoring and design improvement sh to py

> ## Change Summary
> replace sh to Python
> refactoring and design improvement
> ## Types of changes
> * [ ]  Bug fix (non-breaking change which fixes an issue)
> * [ ]  New feature (non-breaking change which adds functionality)
> * [x]  Code style update (formatting, renaming)
> * [ ]  Refactoring (no functional changes)
> * [ ]  Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
> * [ ]  Other (please describe):
> 
> ## Related Task(s)
> *https://phabricator.vyos.net/T4767
> 
> ## Component(s) name
> src/op_mode/generate_ipsec_debug_archive.py
> 
> ## Proposed changes
> ## How to test
> Generate five version of tech-support archive:
> To test please, change the code and file type, then run the command

> ```
>$ generate ipsec debug-archive 
> ```
>check an archive in "/tmp" folder (ipsec_debug_***)
> 
> 
> ## Checklist:
> * [x]  I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
> * [x]  I have linked this PR to one or more Phabricator Task(s)
> * [ ]  I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
> * [ ]  My commit headlines contain a valid Task id
> * [ ]  My change requires a change to the documentation
> * [ ]  I have updated the documentation accordingly
